File: 0001-two-sum/0001-two-sum.py
"""
Time = O(n)
Space = O(n)
"""
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        remain = {}
        
        for i in range(len(nums)):
            if nums[i] not in remain:
                remain[target-nums[i]] = i
            else:
                return [remain[nums[i]], i]
        
        return [-1, -1]

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> umap;
        vector<int> res;

        for (int i = 0; i < nums.size(); i++) {
            if (umap.find(nums[i]) != umap.end()) {
                res.push_back(umap[nums[i]]);
                res.push_back(i);
                return res;
            } else {
                umap[target - nums[i]] = i;
            }
        }
        return res;
    }
};